### PR TITLE
fix(rpc_client): enables gRPC keepalive ping

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Current if meta node drops connection silently (without sending TCP RST), meta_client's channel won't be aware of it, results in

- meta_client's inflight RPC blocks forever, if we don't set a request timeout for it, which is true for now.
- meta_client's server-streaming receiver blocks forever, without receiving data or error, e.g. notification client, compactor client.

This PR enables gRPC keepalive ping, which will throw errors to upper layer if channel is not working.

BTW with regard to request timeout, we cannot simply set a request timeout for the channel, because some requests are expected to take long time, e.g. createMV.

## Checklist

~~- [ ] I have written necessary rustdoc comments~~
~~- [ ] I have added necessary unit tests and integration tests~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by zwang28** [comment]
This PR also removes timeout for heartbeat RPC. I think it's used to handle the same issue here?
@skyzh

File: src/rpc_client/src/meta_client.rs [hunk 1]
```diff
@@ -655,6 +650,10 @@ impl GrpcMetaClient {
         let channel = tokio_retry::Retry::spawn(retry_strategy, || async {
             let endpoint = endpoint.clone();
             endpoint
+                .http2_keep_alive_interval(Duration::from_secs(
```
Comment (skyzh) [inline src/rpc_client/src/meta_client.rs:653]: > If the ping is not acknowledged within the timeout, the connection will be closed. Does nothing if http2_keep_alive_interval is disabled.
> 
> Default is 20 seconds.

https://docs.rs/tonic/latest/tonic/transport/struct.Server.html

**Review by skyzh — APPROVED** [review]
Not sure if this really works. Both params already have default val. We can get this merged as long as someone feels that "it takes effect" 🤣

**Comment by skyzh** [comment]
> This PR also removes timeout for heartbeat RPC. I think it's used to handle the same issue here?

Probably not. We had it because sometimes meta node will stuck (due to deadlock). In this case, the kernel will still actively maintain the TCP connection, but we might still timeout.

File: src/rpc_client/src/meta_client.rs (hunk 1 above)
Comment (zwang28) [reply src/rpc_client/src/meta_client.rs:653]: http2_keep_alive_interval is disabled by default in tonic.

Comment (zwang28) [reply src/rpc_client/src/meta_client.rs:662]: I did a test to verify the difference http2_keep_alive_interval makes:
- establish a stream, halt the server process (via breakpoint).
- when http2_keep_alive_interval is disabled, client won't see any error. When server is resumed, the stream remain available (can receive message).
- when http2_keep_alive_interval is enabled, client will see error after a while, and the stream is ended (client receive None after previous error. So it need to reestablish another one).

**Pull request merged**